### PR TITLE
Development -> Master

### PR DESCRIPTION
**Pull request by mrjones2014**
Adds:

- creating sections
- editing section names
- creating articles
- editing articles
- markdown editor
- markdown preview tab on editor
- syntax highlighting in markdown

### CODE REVIEW

File: Client/wwwroot/blazorContextMenu.css [hunk 1]
```diff
@@ -0,0 +1,204 @@
+﻿.blazor-context-menu--default {
```
Comment (mrjones2014) [inline Client/wwwroot/blazorContextMenu.css:1]: this is from the `Blazor.ContextMenu` package, I had to manually include it

File: Client/wwwroot/blazorContextMenu.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
+﻿"use strict";
```
Comment (mrjones2014) [inline Client/wwwroot/blazorContextMenu.js:1]: this is from the `Blazor.ContextMenu` package, I had to manually include it

File: Client/Components/SidebarEditSection.razor [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    protected override Task OnParametersSetAsync()
+    {
+        Title = InitialTitle;
+        return base.OnParametersSetAsync();
+    }
+
+    protected override void OnAfterRender(bool firstRender)
+    {
+        if (firstRender)
```
Comment (joshuapeters) [inline Client/Components/SidebarEditSection.razor:39]: what's this doing on first render? A comment here would help clear up any confusion

File: Client/Pages/ArticlePage.razor [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    protected override void OnInitialized()
+    {
+        _articleService = new ArticleService(NavigationManager);
+    }
+
+    protected override async Task OnParametersSetAsync()
+    {
+        if (Article == null)
```
Comment (joshuapeters) [inline Client/Pages/ArticlePage.razor]: consider inverting this condition to reduce nesting

File: Client/Pages/Edit.razor [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    protected override void OnInitialized()
+    {
+        _articleService = new ArticleService(NavigationManager);
+    }
+
+    protected override async Task OnParametersSetAsync()
+    {
+        if (Article == null)
```
Comment (joshuapeters) [inline Client/Pages/Edit.razor]: ditto on the inversion here

File: Client/Shared/NavMenu.razor [hunk 6]
```diff
@@ -18,94 +19,277 @@
...
+    private ArticleService    _articleService;
     private bool collapseNavMenu = true;
     private string NavMenuCssClass => collapseNavMenu ? "collapse" : null;
 
-    private long             expandedSectionId       { get; set; }
-    private List<ArticleDto> expandedSectionArticles { get; set; }
-    private bool             isLoadingArticles       { get; set; }
+    private bool _isLoadingAll = false;
+    private long _editingSectionId = -1;
+    private bool _showSectionCreate = false;
```
Comment (joshuapeters) [inline Client/Shared/NavMenu.razor:110]: minor, looks like you were aligning some things and not others.

**Review by joshuapeters — APPROVED** [review]

File: Client/Components/SidebarEditSection.razor (hunk 3 above)
Comment (mrjones2014) [reply Client/Components/SidebarEditSection.razor:39]: it's using JS interop to call a JS function to focus the input element. I'll add a comment.

File: Client/Shared/NavMenu.razor (hunk 6 above)
Comment (mrjones2014) [reply Client/Shared/NavMenu.razor:110]: yeah, the formatter in Rider doesn't seem to understand that the `.razor` extension contains C# code 😬

**Pull request merged**